### PR TITLE
Fix the Rogue DHCP validation

### PR DESCRIPTION
**Pull request by tomassedovic**
The first commit fixes the validation on RHEL by using a virtualenv instead of relying on pip being available (it's not).

That uncovered two issues with reporting errors back to the API which are fixed in the remaining two comments.

You can verify the last comment by running the `ntpstat` validation. Without the commit it just returns "failed" with no explanation. With it, you'll see the command it tried to run and it's stderr.

### CODE REVIEW

**Comment by mandre** [comment]
Now that the `rogue_dhcp.py` file  is not an ansible module anymore, maybe it would be better to move it to the `files` directory.

File: ansible-tests/validations/rogue-dhcp.yaml [hunk 1]
```diff
@@ -7,13 +7,15 @@
...
-  - name: Install scappy
-    pip: name=scapy state=present
-  - name: Look for a rogue DHCP
-    rogue_dhcp: networks="{{ networks }}" timeout_seconds=60
+  - name: Create a temporary directory for a virtualenv
+    command: mktemp -d -t venv-XXXXX
+    register: virtualenv
+  - name: Install scapy
+    pip: name=scapy state=present virtualenv={{ virtualenv.stdout }}
+  - name: Copy the DHCP script over
```
Comment (mandre) [inline ansible-tests/validations/rogue-dhcp.yaml]: I believe it's not necessary to copy the file to the virtualenv's directory. If you call the python binary with the virtualenv's path, then you should normally have all you need to run rogue_dhcp.py.

**Comment by tomassedovic** [comment]
Good point about `files`. I wanted to do that but forgot.

And you're right that the file doesn't have to be copied to the virtualenv, but it should be copied onto the host (even the undercloud validations may be run from another host). And I figured -- we've already created a temp dir, might as well drop it there.

Is there a location you'd like more?

**Comment by mandre** [comment]
You could use the script module (http://docs.ansible.com/ansible/script_module.html) and set the `rogue_dhcp.py`'s shebang to point to your virtualenv's python. I think that would be cleaner than copying the file yourself.

**Comment by tomassedovic** [comment]
That sounds good. We'd have to hardcode the virtualenv path though, wouldn't we? Or did you mean having `rogue_dhcp.py` be a jinja template and set the shebang?

**Comment by mandre** [comment]
Make it a j2 template if you want to be fancy, but I think it's fine just hardcoding it since I don't see the script being used somewhere else.

**Comment by tomassedovic** [comment]
Fair enough :-). Just wanted to be sure what you meant.

**Comment by tomassedovic** [comment]
@mandre updated.

**Pull request merged**